app/models/katello/product_content.rb

```diff
@@ -52,11 +52,24 @@ def self.with_valid_subscription(organization)
       where(:product_id => Katello::PoolProduct.where(:pool_id => organization.pools).select(:product_id))
     end
 
-    # used by Katello::Api::V2::RepositorySetsController#index
+    # following 4 methods used by Katello::Api::V2::RepositorySetsController#index
     def repositories
       Katello::Repository.in_default_view.where(:root_id => product.root_repositories.has_url.where(:content_id => content.cp_content_id))
```

Comment:
Is it possible to drop `has_url` here?

Reply:
`repositories` is used in the existing API response / rabl for repository sets, so I didn't want to change that output.

Reply:
Why repos without url are displayed?

Reply:
You can have a custom repository that doesn't sync to an upstream repo -- for example, if you upload all the rpms manually.